texar/torch/modules/tokenizers/pretrained_bert_tokenizer.py

```diff
@@ -0,0 +1,199 @@
+            all supported models.
+            If None, the model name in :attr:hparams is used.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    _MAX_INPUT_SIZE = {
+        'bert-base-uncased': 512,
+        'bert-large-uncased': 512,
+        'bert-base-cased': 512,
+        'bert-large-cased': 512,
+        'bert-base-multilingual-uncased': 512,
+        'bert-base-multilingual-cased': 512,
+        'bert-base-chinese': 512,
+    }
+    _VOCAB_FILE_NAMES = {'vocab_file': 'vocab.txt'}
+
+    def __init__(self,
+                 pretrained_model_name: Optional[str] = None,
+                 cache_dir: Optional[str] = None,
+                 hparams=None):
+        super().__init__(hparams=hparams)
+
+        self.config = {
+            'tokenize_chinese_chars': self.hparams['tokenize_chinese_chars'],
+            'do_lower_case': self.hparams['do_lower_case'],
+            'do_basic_tokenize': self.hparams['do_basic_tokenize'],
+            'never_split': self.hparams['never_split'],
+        }
+
+        self.load_pretrained_config(pretrained_model_name, cache_dir)
+
+        if self.pretrained_model_dir is not None:
+            vocab_file = os.path.join(self.pretrained_model_dir,
+                                      self._VOCAB_FILE_NAMES['vocab_file'])
+            assert pretrained_model_name is not None
+            if self._MAX_INPUT_SIZE.get(pretrained_model_name):
+                self.max_len = self._MAX_INPUT_SIZE[pretrained_model_name]
+        else:
+            vocab_file = self.hparams['vocab_file']
+            if self.hparams.get('max_len'):
+                self.max_len = self.hparams['max_len']
+
+        if not os.path.isfile(vocab_file):
+            raise ValueError("Can't find a vocabulary file at path "
+                             "'{}".format(vocab_file))
+        self.vocab = load_vocab(vocab_file)
+        self.ids_to_tokens = dict((ids, tok) for tok, ids in self.vocab.items())
+
+        self.do_basic_tokenize = self.hparams['do_basic_tokenize']
+        if self.do_basic_tokenize:
+            self.basic_tokenizer = BasicTokenizer(
+                do_lower_case=self.hparams["do_lower_case"],
+                never_split=self.hparams["never_split"],
+                tokenize_chinese_chars=self.hparams["tokenize_chinese_chars"])
+        self.wordpiece_tokenizer = WordpieceTokenizer(vocab=self.vocab,
+                                                      unk_token=self.unk_token)
+
+    def forward(self, inputs, task, **kwargs):
+        if task == 'text-to-token':
+            return self.tokenize(inputs)
```

Comment:
Methods like `tokenize(..)` and `encode(..)` should not be exposed to users. So make them private methods.